src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,309 @@
+    # rels_mondo_source = [(mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]) for x in rels_mondo_mondo]
+    # rels_mondo_source = [x for x in rels_mondo_source if any([x[0].startswith(y) for y in owned_prefix_map]) and
+    #   any([x[2].startswith(y) for y in owned_prefix_map])]
+    rels_mondo_non_source_temp = set()
+    rels_mondo_non_synced_temp = set()
+    # - rels_mondo_non_source_temp: Parent and/or child not from source
+    # - rels_mondo_non_synced_temp: Parent and/or child Mondo ID exists in mondo.db but not mondo.sssom.tsv
+    for x in rels_mondo_mondo:
+        try:
+            rels_mondo_source.add((mondo_source_map[x[0]], x[1], mondo_source_map[x[2]]))
+        except KeyError:
+            try:
+                rels_mondo_non_source_temp.add((mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]))
+            except KeyError:
+                rels_mondo_non_synced_temp.add(x)
+    t1 = datetime.now()
+    print(f'  Completed in {(t1 - t0).seconds} seconds')
+
+    # Determine hierarchy diferences -----------------------------------------------------------------------------------
+    # - find which edges appear in both Mondo and source, or only in one or the other
+    # TODO: set diffs to produce these categories
+    #    1. SCR direct in source and Mondo
+    #      DONE: 3. SCR is direct in the source, but not at all in Mondo
+    #        10. SCR is excluded in Mondo, but direct in source
+    #      5. SCR is direct in Mondo, and not in the source at all.
+    in_both_set: Set[RELATIONSHIP] = rels_source.intersection(rels_mondo_source)
+    in_source_only_set: Set[RELATIONSHIP] = rels_source.difference(rels_mondo_source)
+    in_mondo_only_set: Set[RELATIONSHIP] = rels_mondo_source.difference(rels_source)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    robot_subheader = [{
+        'subject_mondo_id': 'ID',
+        'subject_mondo_label': 'LABEL',
+        'object_mondo_id': 'SC %',
+        'subject_source_id': '',  # >A oboInOwl:source?
+        'object_source_id': '',  # ?
+        'object_mondo_label': '',  # ?
+    }]
+    # - Add metadata to edges (labels, both source/Mondo IDs)
+    in_both: List[Dict] = _add_edge_metadata(in_both_set, source_data_map)
+    in_mondo_only: List[Dict] = _add_edge_metadata(in_mondo_only_set, source_data_map)
+    in_source_only: List[Dict] = _add_edge_metadata(in_source_only_set, source_data_map)
+
+    # Case 3: SCR is direct in the source, but not at all in Mondo
+    df3 = pd.DataFrame(in_source_only)
+    df3 = df3.sort_values(['object_mondo_id'], ascending=[False])
+    unmigrated_terms = set(df3[df3['object_mondo_id'] == None]['object_source_id'])
+    if len(unmigrated_terms) >= 200:
+        print(f'Warning: {len(unmigrated_terms)} unmigrated source terms found; a higher number than expected.',
```

Comment:
## Warning for >200 unmigrated terms (FYI: @matentzn)
@matentzn Mark this resolved if this is completed to satisfaction.

You asked for a warning if >200 cases, and I think this is what you were talking about. What happens here is that we are iterating over all MondoID::sourceID pairings in `mondo.sssom.tsv`, and we query the source for subclass relationships for those terms. However, sometimes some of the relationships contain object source IDs that have not yet been migrated to Mondo.

Reply:
I would prefer if we could always use logger.warning(), never print, so I can regulate output levels as desired. Else yes, that is it! Feel free to resolve yourself.